omas/machine_mappings/mast.py

```diff
@@ -0,0 +1,322 @@
+        else:
+            print(f'WARNING: pf_active.coil[{channel}].current.data is missing')
+
+
+@machine_mapping_function(__regression_arguments__, pulse=44653)
+def magnetics_hardware(ods, pulse):
+    r"""
+    Load MAST tokamak flux loops and magnetic probes hardware geometry
+
+    :param ods: ODS instance
+    """
+    from omfit_classes.omfit_efund import OMFITmhdin, OMFITnstxMHD
+
+    mhdin = get_support_file(OMFITmhdin, nstx_filenames('mhdin', pulse))
+    mhdin.to_omas(ods, update='magnetics')
+
+    signals = get_support_file(OMFITnstxMHD, nstx_filenames('signals', pulse))
+
+    for k in ods[f'magnetics.flux_loop']:
+        ods[f'magnetics.flux_loop.{k}.identifier'] = str(signals['mappings']['tfl'][k + 1]['mds_name'])
+
+    for k in ods[f'magnetics.b_field_pol_probe']:
+        ods[f'magnetics.b_field_pol_probe.{k}.identifier'] = str(signals['mappings']['bmc'][k + 1]['mds_name'])
+
+
+@machine_mapping_function(__regression_arguments__, pulse=44653)
+def magnetics_floops_data(ods, pulse, server=None, port=None):
+    r"""
+    Load MAST tokamak flux loops flux data
+
+    :param ods: ODS instance
+
+    :param pulse: shot number
+    """
+    from omfit_classes.omfit_efund import OMFITnstxMHD
+    
+    signals = get_support_file(OMFITnstxMHD, nstx_filenames('signals', pulse))
+    tfl_signals = signals['mappings']['tfl']
+
+    client = get_pyuda_client(server=server, port=port)
+
+    for channel in signals['FL']:
+        sig = signals['FL'][channel]['mds_name']
+
+        scale = signals['FL'][channel]['scale']     
+        try:
+            tmp = client.get(sig,pulse)
+            ods[f'magnetics.flux_loop.{channel}.flux.data'] = tmp.data * scale
+            ods[f'magnetics.flux_loop.{channel}.flux.time'] = tmp.time.data
+            ods[f'magnetics.flux_loop.{channel}.flux.validity'] = 0
+        except pyuda.UDAException:
+            ods[f'magnetics.flux_loop.{channel}.flux.validity'] = -2
+
+
+    # handle uncertainties
+    tfl_signals = signals['mappings']['tfl']
+    for channel in range(len(ods['magnetics.flux_loop']) ):
+        if f'magnetics.flux_loop.{channel}.flux.data' in ods:
+            data = ods[f'magnetics.flux_loop.{channel}.flux.data']
+            rel_error = data * tfl_signals[channel + 1]['rel_error']
+            abs_error = tfl_signals[channel + 1]['abs_error']
+            error = np.sqrt(rel_error**2 + abs_error**2)
+            error[np.abs(data) < tfl_signals[channel + 1]['sig_thresh']] = tfl_signals[channel + 1]['sig_thresh']
+            ods[f'magnetics.flux_loop.{channel}.flux.data_error_upper'] = error
+            # 2*pi normalization is done at this stage so that rel_error, abs_error, sig_thresh are consistent with data
+            ods[f'magnetics.flux_loop.{channel}.flux.data'] /= 2.0 * np.pi
+            ods[f'magnetics.flux_loop.{channel}.flux.data_error_upper'] /= 2.0 * np.pi
```

Comment:
Is this the right approach @orso82 ?

Reply:
Removed in 859246a8a214f617334e02c4f4bb32ca14a1ea06